### PR TITLE
Add igxForOf topic

### PR DESCRIPTION
**Pull request by dkamburov**
Related https://github.com/IgniteUI/igniteui-angular/issues/552

### CODE REVIEW

**Comment by dkamburov** [comment]
@mpavlinov I'm good with them, the overview section is great.

File: en/components/for_of.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        ...
+        IgxForOfModule,
+        ...
+    ]
+})
+export class AppModule {}
+```
+
+### Usage
+
```
Comment (mpavlinov) [inline en/components/for_of.md]: @dkamburov We should add 3 sections here for horizontal, vertical and both virtualization.

File: en/components/for_of.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+### Inputs
+
+Below is the list of all inputs that the developers may set to configure the `igxFor` look/behavior:
+| Name | Type | Description |
+| :--- |:--- | :--- |
+| id | string | Unique identifier of the Grid |
+| `igxForOf` | any[] | The data to be virtualized |
+| `igxForScrollOrientation` | string | Specifies the scroll direction - "horizontal" or "vertical", default value is _vertical_ |
+| `igxForScrollContainer` | any | Specifies the container where the helper scrollbars will be contained, it is useful when nesting the directive |
```
Comment (mpavlinov) [inline en/components/for_of.md]: @dkamburov The description is very generic. I don't understand how to use this option. We should make the description better. Let's discuss this tomorrow.

File: en/components/for_of.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+| `igxForContainerSize` | any |  Specifies the container size in pixels. For "vertical" virtualization this is the height of the container. For "horizontal" virtualization this is the width of the container. |
+| `igxForItemSize` | any | Specifies the item size in pixels. For "vertical" virtualization this is the height of the item/row.  |
```
Comment (mpavlinov) [inline en/components/for_of.md]: @dkamburov Is it only applicable for vertical virtualization? If so, we should explicitly state it in the docs.

File: en/components/for_of.md [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+<div class="divider"></div>
+
+### Methods
+
+Here is a list of all public methods exposed by the **igx-for**:
+
+| Signature       | Description                     |
+| :-------------- | :------------------------------ |
+| `scrollNext()`  | Loads the previous chunk of data |
```
Comment (mpavlinov) [inline en/components/for_of.md]: @dkamburov Maybe "nextChunk"/"previousChunk" names are more appropriate. I'll propose it as a change in the API.

File: en/components/for_of.md [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+<div class="divider--half"></div>
+
+### Outputs
+
+A list of the events emitted by the **igx-for**:
+
+| Name              | Description                                                              |
+| :---------------- | :----------------------------------------------------------------------- |
+| _Event emitters_  | _Notify for a change_                                                    |
+| `onChunkLoaded`   | Used when scrolled to emit the loaded chunk of data                      |
```
Comment (mpavlinov) [inline en/components/for_of.md]: @kdinev From what I know we're not using "ed"/"ing" events anymore. We have only one event to keep the event number lower across the product. Should we remove one of the events?

Comment (kdinev) [reply en/components/for_of.md]: @mpavlinov Correct. @rkaraivanov should chip in on what the correct naming should be. I guess `onChunkLoad`

File: en/components/for_of.md (hunk 3 above)
Comment (kdinev) [reply en/components/for_of.md]: Make the wording such that it's clear what it means in horizontal and vertical scenarios.

File: en/components/for_of.md (hunk 5 above)
Comment (MayaKirova) [reply en/components/for_of.md]: @kdinev @mpavlinov We're actually using both events at the moment in the grid. The "ing" event is used to clean some column/row states in certain scenarios, while the "ed" event is used for the keyboard navigation implementation.
From the directive's perspective itself we definitely need an event before the new chunk is actually loaded to provide  the users with the capability implement remote virtualization. If it's absolutely neccessary to move away from "ing"/"ed" naming convention then we could name one onScroll and the other onChunkLoad. Let me know what you think.

File: en/components/for_of.md [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+## API
+
+### Inputs
+
+Below is the list of all inputs that the developers may set to configure the `igxFor` look/behavior:
+| Name | Type | Description |
+| :--- |:--- | :--- |
+| id | string | Unique identifier of the Grid |
+| `igxForOf` | any[] | The data to be virtualized |
+| `igxForScrollOrientation` | string | Specifies the scroll direction - "horizontal" or "vertical", default value is _vertical_ |
```
Comment (MayaKirova) [inline en/components/for_of.md]: We currently don't set a default value for igxForScrollOrientation.

File: en/components/for_of.md (hunk 4 above)
Comment (dkamburov) [reply en/components/for_of.md]: @mpavlinov Actually the behavior is not described correctly, the methods are scrolling by one item in the appropriate direction, so it's probably better to leave the names as they are and change the description

File: en/components/for_of.md [hunk 7]
```diff
@@ -0,0 +1,149 @@
+---
+title: igxFor
+_description: The Ignite UI for Angular product is providing virtual directive igxFor similar to ngFor, but it is creating scrollbars and only renders small chunks of the data.
```
Comment (kdinev) [inline en/components/for_of.md]: Ignite UI for Angular now exposes a virtual igxFor directive similar to ngFor, which virtualizes DOM object rendering by visualizing only the visible chunks of the data in the DOM.

File: en/components/for_of.md [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+_keywords: Ignite UI for Angular, UI controls, Angular widgets, web widgets, UI widgets, Angular, Native Angular Components Suite, Native Angular Controls, Native Angular Components Library, Virtualization, Performance, Virtual directive, Virtual For
+---
+
+## igxFor
+
+<p class="highlight">Display huge amount of repeated items. The directive is creating scrollable containers and renders small chunks of the data. Is is used inside igx-grid and it can be used to build virtual igx-list.</p>
```
Comment (kdinev) [inline en/components/for_of.md]: Virtualizes DOM object rendering by visualizing only the visible chunks of the data in the DOM. DOM objects are reused while the end-user is scrolling. The igx-grid implements the igxFor directive internally, but the directive can also be implemented on any component that needs to render large amounts of data, like the igx-list.

File: en/components/for_of.md [hunk 9]
```diff
@@ -0,0 +1,149 @@
...
+
+```html
+<span #container>
+    <ng-template *igxFor="data"></ng-template>
+</span>
+```
+
+
+
+The **data** property is an array and it provides the virtualized data.
```
Comment (kdinev) [inline en/components/for_of.md]: The date itself is not virtualized, so just call it "the data"

File: en/components/for_of.md [hunk 10]
```diff
@@ -0,0 +1,149 @@
...
+
+
+The directive can be used to virtualize the data in vertical, horizontal or both directions.
```
Comment (kdinev) [inline en/components/for_of.md]: virtualized the DOM?

File: en/components/for_of.md (hunk 5 above)
Comment (kdinev) [reply en/components/for_of.md]: `onChunkPreload` and `onChunkLoad`?

File: en/components/for_of.md [hunk 11]
```diff
@@ -0,0 +1,156 @@
...
+
+### Outputs
+
+A list of the events emitted by the **igx-for**:
+
+| Name              | Description                                                          |
+| :---------------- | :------------------------------------------------------------------- |
+| _Event emitters_  | _Notify for a change_                                                |
+| `onItemLoad`      | Used when scrolled to emit the loaded data item                      |
+| `onItemPreload`   | Used when scrolled to emit the data item which is about to be loaded |
```
Comment (kdinev) [inline en/components/for_of.md]: I'm waiting on agreement on the PR to the main repository to approve the final API.
https://github.com/IgniteUI/igniteui-angular/pull/619

**Pull request merged**